src/Sentry/Internal/BackgroundWorker.cs

```diff
@@ -142,8 +143,10 @@ private async Task DoWorkAsync()
                         }
                     }
 
-                    if (_queue.TryPeek(out var envelope)) // Work with the envelope while it's in the queue
+                    // Work with the envelope while it's in the queue
+                    if (_queue.TryPeek(out var envelope))
                     {
+                        var eventId = envelope.TryGetEventId(_options.DiagnosticLogger);
```

Comment:
Probably should return null if no diagnostic logger to avoid extra parsing?

Reply:
Should we use it on line 160 as well?

Reply:
eventId  is now reused.

but i dont understand "Probably should return null if no diagnostic logger to avoid extra parsing"

Reply:
I meant given that `TryGetEventId` has work to do to get the event id out of the header, then we could skip it if not logging.  Nevermind though, not super important.